larpandoracontent/LArObjects/LArAdaBoostDecisionTree.h

```diff
@@ -0,0 +1,319 @@
+/**
+ *  @file   larpandoracontent/LArObjects/LArAdaBoostDecisionTree.h
+ *
+ *  @brief  Header file for the lar adaptive boosted decision tree class.
+ *
+ *  $Log: $
+ */
+#ifndef LAR_ADABOOST_DECISION_TREE_H
+#define LAR_ADABOOST_DECISION_TREE_H 1
+
+#include "Pandora/AlgorithmTool.h"
+#include "Pandora/StatusCodes.h"
+
+#include "larpandoracontent/LArHelpers/LArMvaHelper.h"
+
+#include <functional>
+#include <map>
+#include <vector>
+
+namespace lar_content
+{
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+/**
+ *  @brief  AdaBoostDecisionTree class
+ */
+class AdaBoostDecisionTree
```

Comment:
Is it possible for AdaBoostDecisionTree and SupportVectorMachine to have a common interface enforced, so that they *must* work within the constraints of the `LArMvaHelper`, rather than just working because you have made them the same? Would probably just need a purely abstract base class.

Reply:
Base class created and used inside MvaHelper.